src/screens/OtherProfile/index.js

```diff
@@ -63,47 +63,42 @@ const OtherProfile = () => {
   const specificIssueRef = React.useRef();
   const flatListRef = React.useRef();
 
+  const [dataMain, setDataMain] = React.useState({});
+  const [, setDataMainBio] = React.useState('');
   const [user_id, setUserId] = React.useState('');
   const [username, setUsername] = React.useState('');
   const [other_id, setOtherId] = React.useState('');
   const [isShowButton, setIsShowButton] = React.useState(false);
   const [opacity, setOpacity] = React.useState(0);
   const [reason, setReason] = React.useState([]);
   const [yourselfId] = React.useState('');
+  const [blockStatus, setBlockStatus] = React.useState({
+    blocked: false,
+    blocker: false
+  });
   const [loadingBlocking, setLoadingBlocking] = React.useState(false);
+  const [postOffset, setPostOffset] = React.useState(0);
 
   const headerHeightRef = React.useRef(0);
   const interactionManagerRef = React.useRef(null);
-  const [mainFeeds, dispatchOtherProfile] = React.useContext(Context).otherProfileFeed;
+  const [otherProfileFeeds, dispatchOtherProfile] = React.useContext(Context).otherProfileFeed;
   const [profile] = React.useContext(Context).profile;
-  const {timer, viewPostTimeIndex} = mainFeeds;
-  const [, dispatch] = React.useContext(Context).feeds;
+  const [mainFeeds, dispatch] = React.useContext(Context).feeds;
+  const [loading, setLoading] = React.useState(false);
+  const [initLoading, setInitLoading] = React.useState(true);
   const [isAnonimity, setIsAnonimity] = React.useState(false);
   const {params} = route;
+  const {feeds} = otherProfileFeeds;
+  const {timer, viewPostTimeIndex} = mainFeeds;
   const [loadingGenerateAnon, setLoadingGenerateAnon] = React.useState(false);
   const [anonProfile, setAnonProfile] = React.useState();
-  const {createSignChat} = useCreateChat();
-
-  const {
-    feeds,
-    otherProfileData: dataMain,
-    isBlocked,
-    isBlocking,
-    isLoading,
-    isProfileFetching,
-    postOffset,
-
-    refetchFeeds,
-    refetchBlockStatus,
-    refetchOtherProfile,
-    setOtherProfileData: setDataMain
-  } = useOtherProfileScreenHooks(params?.data?.other_id, params?.data?.username);
-
+  const {mappingColorFeed} = useCoreFeed();
   const isSignedMessageEnabled = dataMain.isSignedMessageEnabled ?? true;
   const isAnonimityEnabled = dataMain.isAnonMessageEnabled && isSignedMessageEnabled;
+  const {createSignChat} = useCreateChat();
 
   const {onWillSendViewPostTime} = useViewPostTimeHook(dispatch, timer, viewPostTimeIndex);
-  const {fetchNextFeeds} = useFeedPreloadHook(feeds?.length, () => refetchFeeds(postOffset));
+  const {fetchNextFeeds} = useFeedPreloadHook(feeds?.length, () => getOtherFeeds(postOffset));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Several new state variables have been introduced, and the context is being used to get `otherProfileFeeds`, `profile`, and `mainFeeds`. Make sure these new state variables are properly initialized and cleaned up if necessary during component unmounting to prevent memory leaks. Also, ensure that the context values are being used correctly throughout the component.
```diff
+ const [dataMain, setDataMain] = React.useState({});
+ const [, setDataMainBio] = React.useState('');
+ const [user_id, setUserId] = React.useState('');
+ const [username, setUsername] = React.useState('');
+ const [other_id, setOtherId] = React.useState('');
+ const [isShowButton, setIsShowButton] = React.useState(false);
+ const [opacity, setOpacity] = React.useState(0);
+ const [reason, setReason] = React.useState([]);
+ const [yourselfId] = React.useState('');
+ const [blockStatus, setBlockStatus] = React.useState({
+   blocked: false,
+   blocker: false
+ });
+ const [loadingBlocking, setLoadingBlocking] = React.useState(false);
+ const [postOffset, setPostOffset] = React.useState(0);
+ const [otherProfileFeeds, dispatchOtherProfile] = React.useContext(Context).otherProfileFeed;
+ const [profile] = React.useContext(Context).profile;
+ const [mainFeeds, dispatch] = React.useContext(Context).feeds;
+ const [loading, setLoading] = React.useState(false);
+ const [initLoading, setInitLoading] = React.useState(true);
+ const [isAnonimity, setIsAnonimity] = React.useState(false);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->